File: cpp/Day19/interface.cpp
class Calculator : public AdvancedArithmetic {
public:
    int divisorSum(int n) {
        int sum = 0;
        int stop_iter = sqrt(n);
        for(int i = 1; i <= stop_iter; i++){
            if(n % i  == 0){
                sum += i;
                int other_divisor = (n/i);
                if(other_divisor != i){
                    sum += other_divisor;
                }
            }
        }
};

File: python/Day19/interface.py
class Calculator(AdvancedArithmetic):
    def divisorSum(self, n):
        result = 1
        if(n != 1):
            result += n
        for i in range(2, (n // 2 + 1)):
            if n % i == 0:
                result += i;
        return result
